recipes/mono/mono-osx.recipe

```diff
@@ -0,0 +1,149 @@
+        'bin/resgen',
+        'bin/msbuild',
+        'bin/xbuild',
+    ]
+    _mono = [
+        'bin/mono-sgen64',
+        'etc/mono',
+        'lib/mono/gac',
+        'lib/mono/2.0-api',
+        'lib/mono/4.0-api',
+        'lib/mono/4.5',
+        'lib/mono/4.5-api',
+        'lib/mono/msbuild',
+        'lib/mono/xbuild',
+        'lib/mono/xbuild-frameworks',
+        'include/mono-2.0',
+        'lib/libMonoPosixHelper.dylib',
+        'lib/libmonosgen-2.0.1.dylib',
+        'lib/pkgconfig/monosgen-2.pc',
+        'lib/pkgconfig/mono-2.pc',
+        'lib/pkgconfig/mono.pc',
+    ]
+
+    files_bins = ['mono-sgen', 'mono', 'xbuild', 'msbuild']
+    files_libs = ['libmonosgen-2.0', 'libmono-2.0']
+    files_posix_helper = ['%(sdir)s/libMonoPosixHelper%(srext)s']
+    files_config = ['etc/mono/config', 'etc/mono/4.5/machine.config']
+    files_devel = [
+        'bin/gacutil',
+        'include/mono-2.0',
+        'lib/mono/2.0-api',
+        'lib/mono/4.0-api',
+        'lib/mono/4.5',
+        'lib/mono/4.5-api',
+        'lib/mono/gac',
+        'lib/mono/msbuild',
+        'lib/mono/xbuild',
+        'lib/mono/xbuild-frameworks',
+        'lib/pkgconfig/monosgen-2.pc',
+        'lib/pkgconfig/mono-2.pc',
+        'lib/pkgconfig/mono.pc',
+    ]
+
+    def prepare(self):
+        if self.config.target_platform != self.config.platform or self.config.target_platform != Platform.DARWIN:
+            raise InvalidRecipeError()
+
+    def install(self):
+        mono_prefix_dir = '/Library/Frameworks/Mono.framework/Versions/%s' % self.version
+        if not os.path.exists(mono_prefix_dir):
+            m.error("ERROR: Mono not found!")
+            raise FatalError("Mono not found!")
+        for f in self._mono + self._monoscripts:
+            src = os.path.join(mono_prefix_dir, f)
+            dest = os.path.join(self.config.prefix, f)
+            if os.path.isdir(src):
+                shell.copy_dir (src, dest)
+            elif os.path.isfile(src):
+                if not os.path.exists(os.path.dirname(dest)):
+                    os.makedirs(os.path.dirname(dest))
+                shutil.copy(src, dest)
+            else:
+                print "WARNING: %s not found" % f
+
+        #replace mono scripts
+        for f in self._monoscripts:
+            script = os.path.join(self.config.prefix, f)
+            self.inplace_change(script, mono_prefix_dir, self.config.prefix)
+
+        monosgen_link = os.path.join(self.config.prefix, 'lib', 'libmonosgen-2.0.dylib')
+        mono_link = os.path.join(self.config.prefix, 'lib', 'libmono-2.0.dylib')
+
+        shutil.move(os.path.join(self.config.prefix, 'bin', 'mono-sgen64'),
+                    os.path.join(self.config.prefix, 'bin', 'mono-sgen'))
+        shutil.copy(os.path.join(self.config.prefix, 'bin', 'mono-sgen'),
+                    os.path.join(self.config.prefix, 'bin', 'mono'))
+        if(os.path.exists(mono_link)):
+            os.remove(mono_link)
```

Comment:
It's better to do it like this:
```
try:
    os.unlink(mono_link)
except IOError as e:
    pass
```
`os.path.exists` has some weird behaviours with symlinks, and you also protect against the link being modified while you check.